x-pack/plugins/fleet/common/settings/types.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { z } from 'zod';
+
+export type SettingsSection = 'AGENT_POLICY_ADVANCED_SETTINGS';
+
+export interface SettingsConfig {
+  name: string;
+  title: string;
+  description: string;
+  learnMoreLink?: string;
+  schema: z.ZodTypeAny;
+  api_field: {
+    name: string;
+  };
+  // form error state is passed up to the form
+  updateAdvancedSettingsHasErrors?: (hasErrors: boolean) => void;
```

Comment:
I don't really like this to add a function to the `SettingsConfig`, haven't found a better way to pass down this function to the dynamic components.
It is needed to make the submit button disabled if one of the advanced fields are invalid.

<img width="1154" alt="image" src="https://github.com/elastic/kibana/assets/90178898/b53dd2ba-ffd2-4ead-ba92-715c9846003e">